nums/core/array/blockarray.py

```diff
@@ -160,6 +160,30 @@ def squeeze(self):
             new_block_shape.append(b)
         return self.reshape(new_shape, new_block_shape)
 
+    def swapaxes(self, axis1, axis2):
```

Comment:
This appears correct as-is. Let me propose an alternative:
1. Once you have `grid_swap`, create `rarr_src` without flattening the array. `rarr_src` is now a distinct view on the blocks and doesn't need to be copied again. All that's left is to update the blocks themselves.
2. Iterate over the blocks using `for grid_entry in grid_swap.get_entry_iterator()`. You can update the new view of the blocks `rarr_src` directly using `rarr_src[grid_entry] = rarr_src[grid_entry].swapaxes(axis1, axis2)`.
3. `BlockArray` takes an optional array of blocks as a parameter, so you can directly return `BlockArray(grid_swap, self.system, rarr_src)`.

Of course, rename variables accordingly.

Reply:
I changed the code according to the new `Block.swapaxes` function.
This is not exactly the same as what you proposed, but I think this will work the same as well. 
- I made an empty `rarr_src` numpy array and then initialized each entry with `self.blocks[grid_entry].swapaxex`.